pathology/hovernet/evaluation_ignite.py

```diff
@@ -0,0 +1,150 @@
+)
+from monai.handlers import (
+    MeanDice,
+    StatsHandler,
+    CheckpointLoader,
+)
+from monai.utils.enums import HoVerNetBranch
+from monai.apps.pathology.handlers.utils import from_engine_hovernet
+from monai.apps.pathology.engines.utils import PrepareBatchHoVerNet
+from skimage import measure
+
+
+def prepare_data(data_dir, phase):
+    data_dir = os.path.join(data_dir, phase)
+
+    images = list(sorted(
+        glob.glob(os.path.join(data_dir, "*/*image.npy"))))
+    inst_maps = list(sorted(
+        glob.glob(os.path.join(data_dir, "*/*inst_map.npy"))))
+    type_maps = list(sorted(
+        glob.glob(os.path.join(data_dir, "*/*type_map.npy"))))
+
+    data_dicts = [
+        {"image": _image, "label_inst": _inst_map, "label_type": _type_map}
+        for _image, _inst_map, _type_map in zip(images, inst_maps, type_maps)
+    ]
+
+    return data_dicts
+
+
+def run(cfg):
+    if cfg["mode"].lower() == "original":
+        cfg["patch_size"] = [270, 270]
+        cfg["out_size"] = [80, 80]
+    elif cfg["mode"].lower() == "fast":
+        cfg["patch_size"] = [256, 256]
+        cfg["out_size"] = [164, 164]
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    val_transforms = Compose(
+        [
+            LoadImaged(keys=["image", "label_inst", "label_type"], image_only=True),
+            Lambdad(keys="label_inst", func=lambda x: measure.label(x)),
+            CastToTyped(keys=["image", "label_inst"], dtype=torch.int),
+            CenterSpatialCropd(
+                keys="image",
+                roi_size=cfg["patch_size"],
+            ),
+            ScaleIntensityRanged(keys=["image"], a_min=0.0, a_max=255.0, b_min=0.0, b_max=1.0, clip=True),
+            ComputeHoVerMapsd(keys="label_inst"),
+            Lambdad(keys="label_inst", func=lambda x: x > 0, overwrite="label"),
+            CenterSpatialCropd(
+                keys=["label", "hover_label_inst", "label_inst", "label_type"],
+                roi_size=cfg["out_size"],
+            ),
+            CastToTyped(keys=["image", "label_inst", "label_type"], dtype=torch.float32),
+        ]
+    )
+
+    # Create MONAI DataLoaders
+    valid_data = prepare_data(cfg["root"], "valid")
+    valid_ds = CacheDataset(data=valid_data, transform=val_transforms, cache_rate=1.0, num_workers=4)
+    val_loader = DataLoader(
+        valid_ds,
+        batch_size=cfg["batch_size"],
+        num_workers=cfg["num_workers"],
+        pin_memory=torch.cuda.is_available()
+    )
+
+    # initialize model
+    model = HoVerNet(
+        mode=cfg["mode"],
+        in_channels=3,
+        out_classes=cfg["out_classes"],
+        act=("relu", {"inplace": True}),
+        norm="batch",
+        pretrained_url=None,
+        freeze_encoder=False,
+    ).to(device)
+
+    post_process_np = Compose([
+        Activationsd(keys=HoVerNetBranch.NP.value, softmax=True),
+        Lambdad(keys=HoVerNetBranch.NP.value, func=lambda x: x[1: 2, ...] > 0.5)])
```

Comment:
Could you please remind me of why  this post-processing for and why not `AsDiscrete`? thanks

Reply:
My question is that can't `AsDiscrete` satisfy the need here? I'm a bit confused why discretizing only the second channel of NP?

Reply:
Yes, I also didn't understand why they only threshold the second channel. If you think I don't need to keep the same with them, I will update it, thanks!

Reply:
@KumoLiu I see where the issue is. I thought we discussed to make the NP output a binary. It seems that in #5405 @yiheng-wang-nv kept `np_out_channels` as an argument but at least we should set the `np_out_channels=1` by default.
I can update the default for the model and maybe you can change the post process as follow. What do you think?
- softmax => sigmoid
- Lambdad => AsDescreteD

Reply:
@KumoLiu I see where the issue is. I thought we discussed to make the NP output a binary. It seems that in #5405 @yiheng-wang-nv kept `np_out_channels` as an argument but at least we should set the `np_out_channels=1` by default.

I think we should change the default to 1, and the post process as follow. 
- softmax => sigmoid
- Lambdad => AsDescreted

Also changing the default activation to "sigmoid" in `GenerateWatershedMask`.

What do you think?

Reply:
Hi @drbeh
Of course, I could do that but seems it is also hard-coded in `HoVerNetLoss` and may need to change there.
https://github.com/Project-MONAI/MONAI/blob/dd178ea3bcaadbe6e3da964acc3bb4c027383fbf/monai/apps/pathology/losses/hovernet_loss.py#L131-L132
Thanks!